hathor/cli/side_dag.py

```diff
@@ -186,12 +186,8 @@ def is_option(arg_: str) -> bool:
 def _run_side_dag_node(argv: list[str], *, hathor_node_process: Process, conn: 'Connection') -> None:
     """Function to be called by the main process to run the side-dag full node."""
     logger.info('waiting for hathor node to initialize...')
-    if not conn.poll(HATHOR_NODE_INIT_TIMEOUT):
-        logger.critical(
-            f'side-dag node not started because hathor node failed to initialize before {HATHOR_NODE_INIT_TIMEOUT} '
-            f'seconds timeout'
-        )
-        return
+    while not conn.poll(HATHOR_NODE_INIT_WAIT_PERIOD):
+        logger.info('still waiting for hathor node to initialize...')
 
     message = conn.recv()
```

Comment:
Should we add a timeout to this `recv()` call and log `still waiting for hathor node to initialize...`?

Reply:
That's the exact purpose of the `while` loop above, when `conn.poll` returns `True` and it stops, it's guaranteed that there's a message available and therefore `recv()` will return immediately.

Reply:
This has been removed in https://github.com/HathorNetwork/hathor-core/pull/1079

Reply:
Got it. Make sense. Thanks!